src/SignalR/README.md

```diff
@@ -1,52 +1,52 @@
 
 You can watch an introductory presentation here - [ASP.NET Core SignalR: Build 2018](https://www.youtube.com/watch?v=Lws0zOaseIM)
 
-## Documentation
-
 Documentation for ASP.NET Core SignalR can be found in the [Real-time Apps](https://docs.microsoft.com/aspnet/core/signalr/introduction) section of the ASP.NET Core Documentation site.
 
-## TypeScript Version
+## Description
 
-If you are encountering TypeScript definition issues with SignalR, please ensure you are using the latest version of TypeScript to compile your application. If the issue occurs in the latest TypeScript, please let us know.
+This folder contains the server and client implementations for SignalR.
 
-When in doubt, check the version of TypeScript referenced by our [package.json](clients/ts/package.json) file. That version is the minimum TypeScript version expected to work with SignalR.
+The following contains a description of the sub-directories.
 
-## Packages
+`server/Core`: Contains the main server-side implementation of the SignalR protocol and the Hubs API.
+`server/SignalR`: Contains extensions that help make using SignalR easier.
+`server/StackExchangeRedis`: Contains a backplane implementation using StackExchange.Redis.
+`server/Specification.Tests`: Contains a set of tests for users to use when verifying custom implementations of SignalR types.
+`common/Http.Connections.Common`: Contains common types used by both the server and .NET client for the HTTP layer.
+`common/Http.Connections`: Contains the HTTP implementation layer for LongPolling, ServerSentEvents, and WebSockets on the server.
+`common/Protocols.Json`: Contains the Json Hub Protocol implementation using System.Text.Json for the server and .NET client.
+`common/Protocols.MessagePack`: Contains the MessagePack Hub Protocol implementation for the server and .NET client.
+`common/Protocols.NewtonsoftJson`: Contains the Json Hub Protocol implementation using Newtonsoft.Json for the server and .NET client.
+`common/SignalR.Common`: Contains common types used by both the server and .NET client for the SignalR layer.
+`clients/csharp`: Contains the client-side implementation of the SignalR protocol in .NET.
+`clients/java`: Contains the client-side implementation of the SignalR protocol in Java.
+`clients/ts`: Contains the client-side implementation of the SignalR protocol in TypeScript/JavaScript.
 
-You can install the latest released JavaScript client from npm with the following command:
+## Development Setup
 
-```bash
-npm install @microsoft/signalr
-```
+### Build
 
-The `@microsoft/signalr` package (and it's dependencies) require NPM 5.6.0 or higher.
+By default the build will try to build Java and Typescript projects. If you don't want to include those you can pass "-NoBuildJava" and "-NoBuildNodeJS" respectively to the `build` command to skip them. Or "--no-build-java" and "--no-build-nodejs" on MacOS or Linux.
```

Comment:
"`build` command" makes me think of `dotnet build`. How about?

```suggestion
By default, the build script will try to build Java and Typescript projects. If you don't want to include those you can pass "-NoBuildJava" and "-NoBuildNodeJS" respectively to the `build` script to skip them. Or "--no-build-java" and "--no-build-nodejs" on MacOS or Linux.
```

Also, do we really not support both argument syntaxes on all platforms? I feel that we should.